addon/components/polaris-resource-list/bulk-actions.js

```diff
@@ -0,0 +1,535 @@
+          if (containerWidth > 0) {
+            this.set('containerWidth', containerWidth);
+          }
+        }
+
+        if (smallScreenPopoverVisible || largeScreenPopoverVisible) {
+          this.setProperties({
+            smallScreenPopoverVisible: false,
+            largeScreenPopoverVisible: false,
+          });
+        }
+      },
+      50
+    );
+  },
+
+  didInsertElement() {
+    this._super(...arguments);
+
+    let promotedActions = this.get('promotedActions');
+
+    if (promotedActions && promotedActions.length > MAX_PROMOTED_ACTIONS) {
+      warn(
+        `To provide a better user experience. There should only be a maximum of ${MAX_PROMOTED_ACTIONS} promoted actions.`
+      );
+    }
+
+    this.setContainerNode();
+
+    let largeScreenGroupNode = this.element.querySelector(
+      '.Polaris-ResourceList-BulkActions__Group--largeScreen'
+    );
+
+    if (largeScreenGroupNode) {
+      this.addEventListener(largeScreenGroupNode, 'resize', this.handleResize);
+    }
+  },
+
+  didRender() {
+    this._super(...arguments);
+
+    let moreActionsEl = this.element.querySelector(
+      '.Polaris-ResourceList-BulkActions__Popover'
+    );
+
+    let largeScreenButtonsEl = this.element.querySelector(
+      '.Polaris-ResourceList-BulkActions__Group--largeScreen'
+    );
+
+    if (moreActionsEl) {
+      this.setMoreActionsNode(moreActionsEl);
+    }
+
+    if (largeScreenButtonsEl) {
+      this.setLargeScreenButtonsNode(largeScreenButtonsEl);
+    }
+
+    let {
+      actionsCollection,
+      promotedActions,
+      moreActionsNode,
+      addedMoreActionsWidthForMeasuring,
+      largeScreenButtonsNode,
+      containerNode,
+    } = this.getProperties(
+      'actionsCollection',
+      'promotedActions',
+      'moreActionsNode',
+      'addedMoreActionsWidthForMeasuring',
+      'largeScreenButtonsNode',
+      'containerNode'
+    );
+
+    if (promotedActions && !actionsCollection && moreActionsNode) {
+      addedMoreActionsWidthForMeasuring = moreActionsNode.getBoundingClientRect()
+        .width;
+    }
+
+    let bulkActionsWidth = largeScreenButtonsNode
+      ? largeScreenButtonsNode.getBoundingClientRect().width -
+        addedMoreActionsWidthForMeasuring
+      : 0;
+
+    this.set('bulkActionsWidth', bulkActionsWidth);
+
+    if (containerNode) {
+      this.setProperties({
+        containerWidth: containerNode.getBoundingClientRect().width,
+        measuring: false,
+      });
+    }
+  },
+
+  actions: {
+    toggleSmallScreenPopover() {
+      this.set(
+        'smallScreenPopoverVisible',
+        !this.get('smallScreenPopoverVisible')
+      );
+    },
+
+    toggleLargeScreenPopover() {
+      this.set(
+        'largeScreenPopoverVisible',
+        !this.get('largeScreenPopoverVisible')
+      );
+    },
+
+    setSelectMode(val) {
+      let onSelectModeToggle = this.get('onSelectModeToggle');
+
+      if (onSelectModeToggle) {
```

Comment:
Since we default this to a no-op, there isn't really any need for `setSelectMode`, we could just pass `(action onSelectModeToggle)` in the template 🤷‍♂️